packages/@webex/internal-plugin-device/src/device.js

```diff
@@ -439,7 +440,62 @@ const Device = WebexPlugin.extend({
         });
     });
   },
+  /**
+   * Fetches the web devices and deletes the third of them which are not recent devices in use
+   * @returns {Promise<void, Error>}
+   */
+  deleteDevices() {
+    // Fetch devices with a GET request
+    return this.request({
+      method: 'GET',
+      service: 'wdm',
+      resource: 'devices',
+    })
+      .then((response) => {
+        const {devices} = response.body;
+
+        // Filter devices of type 'WEB'
+        const webDevices = devices.filter((item) => item.deviceType === 'WEB');
+
+        const sortedDevices = orderBy(webDevices, [(item) => new Date(item.modificationTime)]);
+
+        // If there are more than two devices, delete the last third
+        if (sortedDevices.length > 2) {
+          const totalItems = sortedDevices.length;
+          const countToDelete = Math.ceil(totalItems / 3);
+          const urlsToDelete = sortedDevices.slice(0, countToDelete).map((item) => item.url);
+
+          // Send DELETE requests for each URL
+          urlsToDelete.forEach((url) => {
+            this.request({
+              uri: url,
+              method: 'DELETE',
+            }).then(() => {
+              this.clear();
```

Comment:
we are calling `this.clear()` as many times as many devices we delete. I see that this is even verified in unit tests, but I don't understand why - shouldn't we be calling it only once?